phoneapp/src/app.js

```diff
@@ -0,0 +1,127 @@
+/*
+Приложение телефонный справочник
+
+Создайте функцию конструктор.
+У данной функции должны быть методы:
+
+Преобразование телефонного номера из формата 0993378130 в (099) 33-78-130
+Проверка, что телефонный номер содержит только числа
+Добавление пользователей в справочник
+Удаление пользователя по имени, фамилии
+Поиск пользователей по имени - отображает всех пользователей с одинаковым именем
+Изменение имени, фамилии, телефонного номера у выбраного пользователя ( здесь должно быть реализовано через this )
+Сортировка пользователей по номеру телефона, фамилии, имени и тд, по любому из свойств пользователя
+Фильтр по указанному свойству*/
+
+
+class PhoneApp{
+  constructor(){
+    this.dataBaseProps = ['id','name','cname','phone'];
+    this.dataBase = [
+          {id:1, name:'Nafe1', cname:'kname1', phone:'1111111111'},
+          {id:2, name:'dame2', cname:'Cname2', phone:'5222222222'},
+          {id:3, name:'hame3', cname:'qnbme3', phone:'3333333333'},
+          {id:4, name:'aame4', cname:'hname4', phone:'744224444'},
+      ]    
+  }
+
+  transformNumber(number) {
+    let _res = '(';
+    let _number = String(number);
+    _res += number.slice(0,3);
+    _res += ') ';
+    _res += number.slice(3,5);
+    _res += '-';
+    _res += number.slice(5,7);
+    _res += '-';
+    _res += number.slice(7,10);
+    return _res;
+  }
+
+
+
+  _prepareStringValue(name) {  
```

Comment:
prepare for what ?

Reply:
Naming should include knowledge about domain layer and ideally be without details of implementation.

But I guess it's just a helper method which should not know anything about domain